Azure/azure-cli
# [AppService] `az functionapp create`: Remove workarounds for Centauri

**Related command**
`az functionapp create`

**Description**
In Centauri Private Preview, we did the following workarounds (which we are removing with this PR):
- sending the managed environment ID attribute in the PUT request when creating a function app
- using send_raw_request to fetch the managed environment ID attribute in the GET function app request

Removing these workarounds resolves the following [issue](https://github.com/Azure/azure-functions-on-container-apps/issues/22) because after the `enabling_additional_properties_sending` call, any `site_config.app_settings.append` statements are ignored, which is why the `APPINSIGHTS_INSTRUMENTATIONKEY` app setting is not added when the `--app-insights-key` param is passed. 

Additionally, we are removing https://github.com/Azure/azure-cli/blob/22feb5a5dcb1e23f0c1f1a444ff99af5c5502b19/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L3828 in this PR because we already add that app setting later in the function here:
 https://github.com/Azure/azure-cli/blob/22feb5a5dcb1e23f0c1f1a444ff99af5c5502b19/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L3930
The only reason why we were adding it in L3828 before is because of the app settings getting ignored with the workarounds as described in the previous paragraph. 
 
**Testing Guide**
1. Create a Centauri function app: `az functionapp create`
2. See that the managed environment ID attribute is not None: `az functionapp show`

**History Notes**
For Private Preview, we were given an exception to use the workarounds listed above but we want to have those removed.
---

This checklist is used to make sure that common guidelines for a pull request are followed.
